### ISSUE TITLE
Don't trigger deprecation warning if some overload is deprecated

### ISSUE THREAD

**Issue by aueelis** [palantir/tslint#4522]
### Bug Report

- __TSLint version__: 5.12.1
- __TypeScript version__: 3.2.4
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
window.location.reload(); // no deprecation warning

const reload = window.location.reload; // deprecation warning is triggered
reload();
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "deprecation": true
  }
}
```

#### Actual behavior
As of typescript 3.2.1, `window.location.reload(forcedReload: boolean)` is deprecated, while the parameterless version is not. When linting the code, a direct call to `reload()` is ok. However, just accessing the function (as is, e.g., done in unit tests) triggers the deprecation warning.

#### Expected behavior
No deprecation warning is triggered in the code above, because no deprecated functionality is used.

**Comment by lgregoire-yseop** [comment]
Hi, I'm faced to the same issue on my project when I'm trying to use functions overloads.
If I want to access the function without explicitly calling it, TSLint returns a warning.
The `deprecation` rule seems to be triggered when at least one overload is tagged as `@deprecated`.

```ts
class Foo {
    /** @deprecated */
    public bar(p1: number): any;
    public bar(p1: string, p2: number): any;
    public bar(...params: any) {
        return params;
    }
}

const myFoo = new Foo();
myFoo.bar(2);             // deprecation warning    => OK
myFoo.bar('2', 2);        // no deprecation warning => OK
const myBar = myFoo.bar;  // deprecation warning    => Unexpected behavior
```

* **TSLint version**: 5.12.1
* **TypeScript version**: 3.3.3
* **Running TSLint via**: CLI

**Comment by pkelleter** [comment]
what is the status here? unfortunately my project is filled with deprecation warnings after the update. 
are there any plans of resolving this soon?
would be awesome :)

**Comment by JoshuaKGoldberg** [comment]
@pkelleter it's accepting PRs, so as soon as someone has code that can fix the issue, we'd be happy to take it in. You're more than welcome to send that PR if you'd like! 😄 

In the meantime, you can always disable the rule.

**Comment by pkelleter** [comment]
Which rule can i disable, please?
Is there a specific rule for that or do I have to disable tslint deprecations in generel then?
Unfortunately there is no way that I can come up with enough spare time to work myself into this project and fix it on my own :(

**Comment by JoshuaKGoldberg** [comment]
To disable it in your configuration: https://palantir.github.io/tslint/usage/configuration/

To disable it with inline comments: https://palantir.github.io/tslint/usage/rule-flags/

**Comment by benjamincharity** [comment]
We got hit with this for every usage of RxJS's `of`. We're currently disabling like so:

```
// tslint:disable-next-line deprecation
of(..)
```

You could also set that rule to `false` in your configuration (we just still wanted to be able to use that rule elsewhere)

---

BTW there is a PR open for this that apparently needs some adjustments: https://github.com/palantir/tslint/pull/4649

**Comment by kerbs17** [comment]
If it's RxJS usage that is causing the warnings, update to their latest as
they removed their deprecated flag for of:
https://github.com/ReactiveX/rxjs/blob/master/CHANGELOG.md#bug-fixes

**Comment by benjamincharity** [comment]
Ahh great find! Thanks!

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
